upcloud/resource_upcloud_loadbalancer_test.go

```diff
@@ -187,3 +188,52 @@ func TestAccUpcloudLoadBalancer(t *testing.T) {
 		},
 	})
 }
+
+func TestAccUpcloudLoadBalancer_HTTPRedirectValidation(t *testing.T) {
+	// These test data files should fail in pre-plan validation. Thus, these tests are run in plan-only mode.
+	testDataE1, err := os.ReadFile("testdata/upcloud_loadbalancer/loadbalancer_e1.tf")
+	if err != nil {
+		t.Fatal(err)
+	}
+	testDataE2, err := os.ReadFile("testdata/upcloud_loadbalancer/loadbalancer_e2.tf")
+	if err != nil {
+		t.Fatal(err)
+	}
+	testDataE3, err := os.ReadFile("testdata/upcloud_loadbalancer/loadbalancer_e3.tf")
+	if err != nil {
+		t.Fatal(err)
+	}
+	testDataE4, err := os.ReadFile("testdata/upcloud_loadbalancer/loadbalancer_e4.tf")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var providers []*schema.Provider
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(&providers),
+		Steps: []resource.TestStep{
+			{
+				Config:      string(testDataE1),
+				ExpectError: regexp.MustCompile("actions block should contain at least one action"),
+				PlanOnly:    true,
```

Comment:
This should quarantee that resources will not be created even if validation does not return an error